uaclient/entitlements/tests/test_livepatch.py

```diff
@@ -0,0 +1,220 @@
+        with mock.patch('uaclient.util.get_platform_info') as m_platform_info:
+            m_platform_info.return_value = 'xenial'
+            op_status, details = entitlement.operational_status()
+        assert op_status == status.INAPPLICABLE
+        assert 'Livepatch is not available for Ubuntu xenial.' == details
+
+    def test_contract_status_unentitled(self):
+        """The contract_status returns NONE when entitled is False."""
+        tmp_dir = self.tmp_dir()
+        cfg = config.UAConfig(cfg={'data_dir': tmp_dir})
+        cfg.write_cache('machine-token', LIVEPATCH_MACHINE_TOKEN)
+        cfg.write_cache('machine-access-livepatch',
+                        {'entitlement': {'entitled': False}})
+        entitlement = LivepatchEntitlement(cfg)
+        self.assertEqual(status.NONE, entitlement.contract_status())
+
+
+class TestLivepatchProcessDirectives(TestCase):
+
+    @params(('remoteServer', 'remote-server=%s'), ('caCerts', 'ca-certs=%s'))
+    def test_process_directives_call_livepatch_config_command(
+            self, directive_key, livepatch_param_tmpl):
+        """Livepatch config directives are passed to livepatch config."""
+        directive_value = '%s-value' % directive_key
+        cfg = {'entitlement': {'directives': {directive_key: directive_value}}}
+        with mock.patch('uaclient.util.subp') as m_subp:
+            process_directives(cfg)
+        expected_subp = mock.call(
+            ['/snap/bin/canonical-livepatch', 'config',
+             livepatch_param_tmpl % directive_value], capture=True)
+        assert [expected_subp] == m_subp.call_args_list
+
+    @params({}, {'otherkey': 'othervalue'})
+    def test_process_directives_ignores_other_or_absent(self, directives):
+        """Ignore empty or unexpected directives and do not call livepatch."""
+        cfg = {'entitlement': {'directives': directives}}
+        with mock.patch('uaclient.util.subp') as m_subp:
+            process_directives(cfg)
+        assert 0 == m_subp.call_count
+
+
+class TestLivepatchEntitlementCanEnable(TestCase):
+
+    @mock.patch('uaclient.util.is_container', return_value=False)
+    @mock.patch('os.getuid', return_value=0)
+    def test_can_enable_true_on_entitlement_inactive(
+            self, m_getuid, m_is_container):
+        """When operational status is INACTIVE, can_enable returns True."""
+        tmp_dir = self.tmp_dir()
+        cfg = config.UAConfig(cfg={'data_dir': tmp_dir})
+        cfg.write_cache('machine-token', LIVEPATCH_MACHINE_TOKEN)
+        cfg.write_cache('machine-access-livepatch',
+                        LIVEPATCH_RESOURCE_ENTITLED)
+        entitlement = LivepatchEntitlement(cfg)
+        with mock.patch('sys.stdout', new_callable=StringIO) as m_stdout:
+            self.assertTrue(entitlement.can_enable())
+        self.assertEqual('', m_stdout.getvalue())
+
+    @mock.patch('uaclient.util.is_container', return_value=True)
+    @mock.patch('os.getuid', return_value=0)
```

Comment:
It's not particularly important, but as you don't use `m_getuid` in either of these tests, you could avoid it being passed in at all by invoking the decorator like this:

```py
@mock.patch('os.getuid', Mock(return_value=0))
```

Also, given that it's on each test in this class, you could also move it to be a class decorator; it would then apply to any additional tests written.

Reply:
ahh right, good point moved to class decorator.

Reply:
held off on the mock.Mock for the moment.